test_framework/script.py

```diff
@@ -1219,12 +1230,18 @@ def __init__(self, left=None, right=None):
         self.left = left
         self.right = right
 
-    # TODO: Not nice, but seems to be necessary for sorting by priority queue.
+    def tagged_hash(self):
```

Comment:
I think you can replace this whole function with something like:

```
def tagged_hash(self):
    # lexicographically sort the child hashes, concat them and hash with "TapBranch" tagged hash
    return TaggedHash("TapBranch", b''.join(sorted([h_l,h_r])))
```

Reply:
Yes, this is much nicer. Thank you!